app/models/membership_application.rb

```diff
@@ -74,20 +73,8 @@ class MembershipApplication < ApplicationRecord
       transitions from: :waiting_for_applicant, to: :ready_for_review
     end
 
-    event :ask_for_payment do
-      transitions from: :under_review, to: :waiting_for_payment
-    end
-
-    event :cancel_waiting_for_payment do
-      transitions from: :waiting_for_payment, to: :under_review
-    end
-
-    event :received_payment do
-      transitions from: :waiting_for_payment, to: :accepted, after: :accept_membership
-    end
-
     event :accept do
-      transitions from: [:under_review, :rejected], to: :accepted, guard: [:paid?, :not_a_member?], after: :accept_membership
+      transitions from: [:under_review, :rejected], to: :accepted, after: :accept_membership
```

Comment:
I know it's not likely to ever be triggered, but I'd like to add `guard: :not_a_member?` back to the transition here,

and similarly in the `event :reject do...`  I'd like to add `guard: :is_member?` to
```
transitions from: [:under_review, :accepted], to: :rejected, after: :reject_membership
```

It makes the intent of those events and transitions very clear to anyone trying to understand this later.

FYI for those who haven't worked with AASM much,
`guard: <one method name or an array of them>`  is the same as an `if` clause, i.e. you could also say

`transitions from: [:under_review, :rejected], to: :accepted, after: :accept_membership if :not_a_member?`

Reply:
Hm. I left them out deliberately because of the two cases where an application can be reviewed while the applicant already is a member: 1 - a second application (with a different company) & 2 -  an application used to prolong/review the membership.